src/OrleansConsulUtils/ConsulBasedMembershipTable.cs

```diff
@@ -0,0 +1,215 @@
+﻿using Consul;
+using Orleans.Messaging;
+using Orleans.Runtime.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Orleans.Runtime.Host
+{
+    /// <summary>
+    /// A Membership Table implementation using Consul 0.6.0  https://consul.io/
+    /// </summary>
+    public class ConsulBasedMembershipTable : IMembershipTable, IGatewayListProvider
+    {
+        //Consul does not support the extended Membership Protocol and will always return the same table version information
+        private readonly TableVersion _tableVersion = new TableVersion(0, "0");
+
+        private TraceLogger _logger;
+        private ConsulClient _consulClient = null;
+        private String _deploymentId;
+        private String _connectionString;
+
+        private TimeSpan _maxStaleness;
+
+        public TimeSpan MaxStaleness
+        {
+            get { return _maxStaleness; }
+        }
+
+        public Boolean IsUpdatable
+        {
+            get { return true; }
+        }
+
+        public Task InitializeGatewayListProvider(ClientConfiguration config, TraceLogger traceLogger)
+        {
+            _logger = traceLogger;
+            _deploymentId = config.DeploymentId;
+            _connectionString = config.DataConnectionString;
+            _maxStaleness = config.GatewayListRefreshPeriod;
+
+            _consulClient =
+                new ConsulClient(
+                    new ConsulClientConfiguration
+                    {
+                        Address = new Uri(config.DataConnectionString)
+                    });
+
+            return TaskDone.Done;
+        }
+
+        /// <summary>
+        /// Initializes the Consul based membership table.
+        /// </summary>
+        /// <param name="config">The configuration for this instance.</param>
+        /// <param name="tryInitPath"></param>
```

Comment:
should be called `tryInitTableVersion`. I know you don't use it, but just for cleanness.